src/Data/AudienceTree.elm

```diff
@@ -0,0 +1,361 @@
+module Data.AudienceTree exposing
+    ( AudienceTree
+    , DuplicatedIdFolder
+    , Errors
+    , audiences
+    , children
+    , fromAudiencesAndFolders
+    , name
+    )
+
+import Data.Audience exposing (Audience)
+import Data.AudienceFolder exposing (AudienceFolder)
+import Dict exposing (Dict)
+
+
+type AudienceTree
+    = AudienceTree AudienceTreeInternal
+
+
+type alias AudienceTreeInternal =
+    { name : String
+    , children : List AudienceTree
+    , audiences : List String
+    }
+
+
+children : AudienceTree -> List AudienceTree
+children (AudienceTree a) =
+    a.children
```

Comment:
Nitpick: The name `children` was a bit confusing to me when I read it in the `String.fromInt <| List.length (Tree.children tree) + List.length (Tree.audiences tree)` snippet in `Main`. Maybe a name like `childTrees` might describe it better - make it clear that no audiences are present in the result value?

Reply:
Good catch, you're right!

I'm not really happy with the wording with this module. Maybe `subFolders` might be even better than `childTrees`.